mesonbuild/backend/ninjabackend.py

```diff
@@ -3699,8 +3695,8 @@ def generate_ending(self) -> None:
                 # Add the first output of each target to the 'all' target so that
                 # they are all built
                 #Add archive file if shared library in AIX for build all.
-                if isinstance(t, build.SharedLibrary):
-                    if self.environment.machines[t.for_machine].is_aix():
+                if self.environment.machines[t.for_machine].is_aix():
+                    if isinstance(t, build.SharedLibrary) and t.aix_so_archive:
```

Comment:
Can you flip this around? it should be the same except matching the original order of conditions:

```suggestion
                if isinstance(t, build.SharedLibrary) and t.aix_so_archive:
                    if self.environment.machines[t.for_machine].is_aix():
```

Reply:
Done